src/operations/txbuild.js

```diff
@@ -1071,84 +968,93 @@ function makeNameImport(name: string,
  * @param {String} tokenType - the type of tokens to send
  * @param {Object} tokenAmount - the BigInteger encoding of an unsigned 64-bit number of
  *  tokens to send
  * @param {String} scratchArea - an arbitrary string to include with the transaction
  * @param {String | TransactionSigner} senderKeyIn - the hex-encoded private key to send
  *   the tokens
+ * @param {String | TransactionSigner} btcFunderKeyIn - the hex-encoded private key to fund
+ *   the bitcoin fees for the transaction. Optional -- if not passed, will attempt to
+ *   fund with sender key.
+ * @param {boolean} buildIncomplete - optional boolean, defaults to false,
+ *   indicating whether the function should attempt to return an unsigned (or not fully signed)
+ *   transaction. Useful for passing around a TX for multi-sig input signing.
  * @returns {Promise} - a promise which resolves to the hex-encoded transaction.
  * This function does not perform the requisite safety checks -- please see the
  * safety module for those.
  * @private
  */
 function makeTokenTransfer(recipientAddress: string, tokenType: string,
                            tokenAmount: BigInteger, scratchArea: string,
-                           senderKeyIn: string | TransactionSigner) {
+                           senderKeyIn: string | TransactionSigner,
+                           btcFunderKeyIn?: string | TransactionSigner,
+                           buildIncomplete?: boolean = false) {
   const network = config.network
+  const separateFunder = !!btcFunderKeyIn
 
-  let senderKey
-  if (typeof senderKeyIn === 'string') {
-    senderKey = PubkeyHashSigner.fromHexString(senderKeyIn)
-  } else {
-    senderKey = senderKeyIn
-  }
+  const senderKey = getTransactionSigner(senderKeyIn)
+  const btcKey = btcFunderKeyIn ? getTransactionSigner(btcFunderKeyIn) : senderKey
 
-  const txPromise = Promise.all([network.getConsensusHash()])
-    .then(([consensusHash]) =>  makeTokenTransferSkeleton(
+  const txPromise = network.getConsensusHash()
+    .then(consensusHash =>  makeTokenTransferSkeleton(
       recipientAddress, consensusHash, tokenType, tokenAmount, scratchArea))
 
-  return senderKey.getAddress().then(senderAddress =>
-    Promise.all([network.getUTXOs(senderAddress), network.getFeeRate(), txPromise])
-      .then(([utxos, feeRate, tokenTransferTX]) => {
-        const txB = bitcoinjs.TransactionBuilder.fromTransaction(tokenTransferTX, network.layer1)
-        const signingTxB = fundTransaction(txB, senderAddress, utxos, feeRate, 0)
-        let signingPromise = Promise.resolve()
-        for (let i = 0; i < signingTxB.__tx.ins.length; i++) {
-          signingPromise = signingPromise.then(
-            () => senderKey.signTransaction(signingTxB, i))
-        }
-        return signingPromise.then(() => signingTxB)
-      }))
-    .then(signingTxB => signingTxB.build().toHex())
+  return Promise.all([senderKey.getAddress(), btcKey.getAddress()])
```

Comment:
This is a pretty significant change to this function, but I see no unit test code for it.  Can you add a unit test to check that a `TOKEN_TRANSFER` transaction can be successfully created with a different `btcFunderKeyIn` than the `senderKeyIn`?

Reply:
Yep -- added unit test for this now.